independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/multipart/EntityPartImpl.java

```diff
@@ -0,0 +1,72 @@
+package org.jboss.resteasy.reactive.server.multipart;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.Optional;
+
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.*;
+
+import org.jboss.resteasy.reactive.common.util.UnmodifiableMultivaluedMap;
+
+public class EntityPartImpl implements EntityPart {
+    private final String name;
+    private final MultivaluedMap<String, String> headers;
+    private final MediaType mediaType;
+    private final String fileName;
+    private final InputStream content;
+
+    public EntityPartImpl(FormValue value, String controlName) {
+        this.name = controlName;
+        this.headers = new UnmodifiableMultivaluedMap(value.getHeaders());
+        this.mediaType = MediaType.valueOf(value.getHeaders().getFirst("Content-Type"));
+        this.fileName = value.getFileName();
+
+        try {
+            if (value.isFileItem()) {
+                this.content = new FileInputStream(value.getFileItem().getFile().toFile());
+            } else {
+                this.content = new ByteArrayInputStream(value.getValue().getBytes(Charset.defaultCharset()));
```

Comment:
I'm not sure what encoding the byte array should have, but I suspect not the default charset. First, because we prefer utf-8 everywhere, and second because it's possible that anybody reading the `InputStream` will use the part's encoding as set in headers or media type, which would be different to the default charset.

Also, perhaps the `InputStream` should be build lazily in `getContent`? Otherwise if a user closes it, doing two calls to `getContent` will fail.